### PR TITLE
Migrate from BodyGenerator to DynamicBodySource

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: http-client/src/main/java/com/proofpoint/http/client/balancing/RetryingResponseHandler.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ public Boolean call()
                     exception);
```
Comment (gwittel) [inline http-client/src/main/java/com/proofpoint/http/client/balancing/RetryingResponseHandler.java]: (lines 72-74).  Missed this last time.  May be good to put  a note that we're intentionally leaving the stack trace out.

Comment (johngmyers) [inline http-client/src/main/java/com/proofpoint/http/client/balancing/RetryingResponseHandler.java]: That was pull #387

File: CHANGES [hunk 2]
```diff
@@ -1,5 +1,38 @@
...
+  - We changed the reporting client to use DynamicBodySource.
+
+* Events
+
+  - We deprecated EventClient.post(EventGenerator) and EventGenerator.
+    This interface effectively required implementation on top of
+    BodyGenerator. No replacement is currently planned.
+  - We changed the HttpEventClient.post(Iterable) and .post(T...)
+    implementations to use DynamicBodySource.
+
```
Comment (gwittel) [inline CHANGES]: Missing a note that the reporting client has also been changed to use the ```DynamicBodySource``` as well

**Comment by gwittel** [comment]
LGTM.

Its big enough that its hard to feel like I covered everything well though.

**Pull request merged**